providers/base/bin/cpuid.py

```diff
@@ -139,6 +140,26 @@
         "Whisky Lake":      ['0x806eb', '0x806ec'],
         }
 
+# Server chips use different generations, so skip them
```

Comment:
What do Server chips use then? Especially since the IDs for Westmere, Sandy Bridge, Ivy Bridge, Haswell, Broadwell, and Skylake all express server CPUs (Xeons of various flavors).

Also, one project I had for this cycle is adding in information like kernel version where each is supported initially.  The reason I mention that is the idea I had was to change the CPUID dict to use a dict of dicts like this:

```
"Sapphire Rapids":  {"kernel": ['5.15']; "cpuid": ['0x806f3', '0x806f6', '0x806f7', '0x806f8']},
```
So we could do things like run this to determine if the kernel being used is the minimal required kernel for the CPU identified (and thus fail if someone runs a test on Sapphire Rapids using 5.4, for example).

Would it be better to use your changes here and create the dict of dicts mentioned above so that it's all together rather than in separate lookup tables?

Reply:
The server chips started a new generation branding a few years back: 
Skylake (1st Generation Intel® Xeon® Scalable Processors)
Cascade Lake (2nd Generation Intel® Xeon® Scalable Processors)
Ice Lake (3rd Generation Intel® Xeon® Scalable Processors)
Sapphire Rapids (4th Generation Intel® Xeon® Scalable Processors)

Someone with a bit deeper history on generation numbering of Xeon products might be able to add to this, but I'm not familiar with a version numbering system before that. I specifically didn't add Xeon CPUs because I'm not sure how relevant this is for servers. For desktop/mobile chips, the generation is definitely referenced a lot to determine whether or not certain features get enabled.

For informational purposes, this script was probably doing enough to just give a less specific generation (i.e. Ice Lake instead of Ice Lake-D or even Ice Lake Xeon), but adding something more functional seems like it might cause some problems with the two types of being intermingled. I didn't realize that these would need to be disentangled, but it seems like a good idea before merging this. It sounds like you'd like to see this for Xeon too, but my thinking is that it should be a different flag so that people get what they expect. For instance, if someone intends to get something Ice Lake Xeon ("3" for servers) or higher, and they get "6" back from the script for Skylake, that test will not be running as intended. I think I'd rather their test fail because they used the desktop CPU gen flag and no desktop CPU was found. 

As for the kernel version addition, that feels like a nice change vs separate lookup tables. I can switch it around a bit

Reply:
The server chips started a new generation branding a few years back:

Even older than that:
Ivy Bridge = Intel Xeon v2
Haswell = Intel Xeon v3
Broadwell = Intel Xeon v4

Then you get to skylake which is confusing because
Skylake Xeon E3 = v6
Skylake Scalable = Xeon Scalable 1st Gen

Heh... that was a transition one though, thankfully, after that they stopped versioning the smaller Xeons.